misk-hibernate/src/main/kotlin/misk/hibernate/EntityGroupTransactionOrder.kt

```diff
@@ -0,0 +1,44 @@
+package misk.hibernate
+
+/**
+ * Vitess multi shard transactions are guaranteed to be committed in the order that updates are
+ * issued. So if you have entity groups A and B, you issue an update against A followed by an
+ * update to B, then the transaction against the shard of A will be fully committed before the
+ * transaction against the shard of B. You can use this fact to make cross shard transactions safe
+ * by committing clean up tasks to the earlier shards.
+ */
+interface EntityGroupTransactionOrder {
```

Comment:
i'm having a hard time following how this works since this does not have an impl.

Reply:
Yep, please ignore this PR for now .. implementation coming!